### PR TITLE
Compatibility with dplyr 1.0.9

### PR DESCRIPTION
**Pull request by krlmlr**
and dbplyr update.

This is a breaking change:

- Adding required `conflict` or `unmatched` arguments
- Removing unimplemented `check` argument
- Removed message when `in_place` is missing
- Unexported `sql_rows_*()` methods
- Ready for dbplyr > 2.1.1, using dbplyr methods if provided, without breaking tests
- Added `Matching, by = ...` message when `by` is omitted (with dbplyr > 2.1.1)
- New `rows_append()` and `dm_rows_append()`
- Fix detection of availability of primary keys in `dm_rows_*()` operations

### CODE REVIEW

File: R/rows-db.R [hunk 1]
```diff
@@ -62,9 +57,13 @@ NULL
...
 #' @rdname rows-db
 rows_insert.tbl_dbi <- function(x, y, by = NULL, ...,
-                                in_place = NULL, copy = FALSE, check = NULL,
+                                in_place = FALSE,
+                                conflict = NULL,
+                                copy = FALSE,
                                 returning = NULL) {
 
+  stopifnot(identical(conflict, "ignore"))
+
```
Comment (moodymudskipper) [inline R/rows-db.R:70]: Should our doc explain that (and why) only "ignore" is supported for `conflict` and `unmatched` ?

Comment (krlmlr) [reply R/rows-db.R:70]: This is only compatibility code that will be removed as soon as dbplyr 2.2.0 hits CRAN. No need to spend time here, I guess.

File: R/rows-db.R [hunk 2]
```diff
@@ -292,11 +305,6 @@ rows_upsert.tbl_dbi <- function(x, y, by = NULL, ...,
   name <- target_table_name(x, in_place)
 
   if (!is_null(name)) {
-    # Checking optional, can rely on primary key constraint
-    if (is_true(check)) {
-      check_db_superset(x, y, by)
-    }
-
     if (is_empty(new_columns)) {
       return(invisible(x))
```
Comment (moodymudskipper) [inline R/rows-db.R:304]: Are we sure we want to return sometimes visibly sometimes invisibly ?

File: R/aaa-dbplyr.R [hunk 3]
```diff
@@ -1,3 +1,16 @@
+replace_if_dbplyr_has <- function(fun) {
+  dbplyr_ns <- asNamespace("dbplyr")
+
+  fun <- as_string(ensym(fun))
+  value <- mget(fun, dbplyr_ns, mode = "function", ifnotfound = list(NULL))[[1]]
+  if (!is.null(value)) {
+    assign(fun, value, inherits = TRUE)
+    "dbplyr"
+  } else {
+    "dm"
```
Comment (moodymudskipper) [inline R/aaa-dbplyr.R:14]: should side effect functions return invisibly ?

File: R/aaa-dbplyr.R [hunk 4]
```diff
@@ -1,4 +1,8 @@
 replace_if_dbplyr_has <- function(fun) {
+  if (!requireNamespace("dbplyr", quietly = TRUE)) {
+    return()
```
Comment (moodymudskipper) [inline R/aaa-dbplyr.R:3]: `return(invisible())` ?

File: R/aaa-dbplyr.R [hunk 5]
```diff
@@ -12,15 +16,18 @@ replace_if_dbplyr_has <- function(fun) {
...
 
-  if (is_empty(methods)) {
-    return()
+    # Register our method implementations only if dbplyr doesn't provide them
+    dbplyr_methods <- mget(names(methods), dbplyr_ns, mode = "function", ifnotfound = list(NULL))
+
+    methods <- methods[map_lgl(dbplyr_methods, is.null)]
+
+    if (is_empty(methods)) {
+      return()
```
Comment (moodymudskipper) [inline R/aaa-dbplyr.R:29]: invisible ?

File: R/rows-db.R (hunk 2 above)
Comment (krlmlr) [reply R/rows-db.R:304]: For `in_place = TRUE` we do return invisibly, because this is an operation with side effects.

File: R/aaa-dbplyr.R (hunk 4 above)
Comment (krlmlr) [reply R/aaa-dbplyr.R:3]: You're right, but it's not too important here and below, because this is an internal function called in `.onLoad()` .

**Comment by krlmlr** [comment]
@moodymudskipper: Please approve when it's "good enough".

**Review by moodymudskipper — APPROVED** [review]

**Pull request merged**